# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsGlobalWindowInner.cpp
# Commit: f8a2a169b296
# Full Hash: f8a2a169b2969d29e04733ae27a56459e5359973
# Author: Alexandru Marc <amarc@mozila.com>
# Date: 2025-05-16 09:19:19
# Regressor Bug: 1550462
# File Overlap Count: 1
# Description:
#   Revert "Bug 1550462 - part 3: Enable `pointerrawupdate` event r=smaug" for causing mochitest failures @ test_pointerrawupdate_event_count.html
#   
#   This reverts commit eb27445e0352d6b4df540adb40e21f72fc052e80.
#   
#   Revert "Bug 1550462 - part 2: Make `PresShell::HandleEvent` dispatch preceding `pointerrawupdate` event r=smaug,dom-core,edgar"
# ==============================================================================

diff -r 8d25773fc82f -r f8a2a169b296 dom/base/nsGlobalWindowInner.cpp
--- a/dom/base/nsGlobalWindowInner.cpp	Fri May 16 06:20:47 2025 +0000
+++ b/dom/base/nsGlobalWindowInner.cpp	Fri May 16 09:30:09 2025 +0300
@@ -1133,8 +1133,6 @@
   }
   StartDying();
 
-  ClearHasPointerRawUpdateEventListeners();
-
   if (mDoc && mDoc->GetWindowContext()) {
     // The document is about to lose its window, so this is a good time to send
     // our page use counters.
@@ -7689,26 +7687,6 @@
   return mTrustedTypePolicyFactory;
 }
 
-void nsPIDOMWindowInner::MaybeSetHasPointerRawUpdateEventListeners() {
-  if (HasPointerRawUpdateEventListeners() || !IsSecureContext()) {
-    return;
-  }
-  mMayHavePointerRawUpdateEventListener = true;
-  if (BrowserChild* const browserChild = BrowserChild::GetFrom(this)) {
-    browserChild->OnPointerRawUpdateEventListenerAdded(this);
-  }
-}
-
-void nsPIDOMWindowInner::ClearHasPointerRawUpdateEventListeners() {
-  if (!HasPointerRawUpdateEventListeners()) {
-    return;
-  }
-  mMayHavePointerRawUpdateEventListener = false;
-  if (BrowserChild* const browserChild = BrowserChild::GetFrom(this)) {
-    browserChild->OnPointerRawUpdateEventListenerRemoved(this);
-  }
-}
-
 nsIURI* nsPIDOMWindowInner::GetDocumentURI() const {
   return mDoc ? mDoc->GetDocumentURI() : mDocumentURI.get();
 }
@@ -7799,7 +7777,29 @@
 
 nsPIDOMWindowInner::nsPIDOMWindowInner(nsPIDOMWindowOuter* aOuterWindow,
                                        WindowGlobalChild* aActor)
-    : mOuterWindow(aOuterWindow), mWindowGlobalChild(aActor) {
+    : mMutationBits(0),
+      mIsDocumentLoaded(false),
+      mIsHandlingResizeEvent(false),
+      mMayHaveDOMActivateEventListeners(false),
+      mMayHaveTouchEventListener(false),
+      mMayHaveSelectionChangeEventListener(false),
+      mMayHaveFormSelectEventListener(false),
+      mMayHaveMouseEnterLeaveEventListener(false),
+      mMayHavePointerEnterLeaveEventListener(false),
+      mMayHaveTransitionEventListener(false),
+      mMayHaveSMILTimeEventListener(false),
+      mMayHaveBeforeInputEventListenerForTelemetry(false),
+      mMutationObserverHasObservedNodeForTelemetry(false),
+      mOuterWindow(aOuterWindow),
+      mWindowID(0),
+      mHasNotifiedGlobalCreated(false),
+      mMarkedCCGeneration(0),
+      mHasTriedToCacheTopInnerWindow(false),
+      mNumOfIndexedDBDatabases(0),
+      mNumOfOpenWebSockets(0),
+      mEvent(nullptr),
+      mWindowGlobalChild(aActor),
+      mWasSuspendedByGroup(false) {
   MOZ_ASSERT(aOuterWindow);
   mBrowsingContext = aOuterWindow->GetBrowsingContext();
 